src/software/new_geom/rectangle.h

```diff
@@ -108,3 +108,5 @@ class Rectangle : public ConvexPolygon
 
     bool operator==(const Rectangle &p) const;
 };
+
+std::ostream &operator<<(std::ostream &os, const Rectangle &rect);
```

Comment:
This should be probably be implemented in `Polygon` and list all the points.

Reply:
done